src/ivio/detail/stream_reader.h

```diff
@@ -28,7 +28,7 @@ class stream_reader {
     auto operator=(stream_reader&&) -> stream_reader& = delete;
 
     size_t read(std::ranges::sized_range auto&& range) const {
```

Comment:
It seems like this should look more like

```cpp
    size_t read(std::ranges::contiguous_range auto&& range) const {
        static_assert(std::same_as<std::ranges::range_value_t<decltype(range)>, char>);
        return stream.rdbuf()->sgetn(std::ranges::data(range), std::ranges::size(range));
    }
```

We access `data()`, so it needs to be `std::ranges::contiguous_range`.
And we expect the range to be over `char`, such that the pointer is `char *`.
`static_assert(sizeof(std::ranges::range_value_t<range_t>) == 1);` might also work, but I think only `char` is used. 
Maybe the check for the value type isn't really needed, because `char *` wouldn't bind to `T *` without an `reinterpret_cast`.

This would also apply to the other `read` functions.